sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -86,16 +86,17 @@ private[speedy] sealed abstract class SBuiltinPure(arity: Int) extends SBuiltinF
     * execution is immediate.
     *
     * @param args arguments for executing the pure builtin
+    * @param machine the Speedy machine (machine state may be modified by the builtin)
     * @return the pure builtin's resulting value (wrapped as a Control value)
     */
-  private[speedy] def executePure(args: ArraySeq[SValue]): SValue
+  private[speedy] def executePure(args: ArraySeq[SValue], machine: Machine[_]): SValue
 
-  override private[speedy] final def execute[Q](
+  private[speedy] final override def execute[Q](
       args: ArraySeq[SValue],
       machine: Machine[Q],
-  ): Control.Value = {
-    Control.Value(executePure(args))
-  }
+  ): Control[Q] =
```

Comment:
Other than "matching" the override type, is there a reason for changing this to `Control[Q]`? Asking as I thought in Scala it was generally better to use specific return types and rely on co-variance to infer the more general type?